src/bundles/pinning.js

```diff
@@ -55,23 +62,83 @@ const uniqueCidBatches = (arrayOfCids, size) => {
   return result
 }
 
+const remotePinLs = (ipfs, params) => {
+  const backoffs = readSetting('remotesServicesBackoffs') || {}
+  const { service } = params
+
+  const { lastTry, tryAfter } = backoffs[service] || { lastTry: 0, tryAfter: PIN_CHECK_INTERVAL }
+  if (lastTry + tryAfter > new Date().getTime()) {
+    throw new Error('still within back-off period')
+  }
+
+  try {
+    return ipfs.pin.remote.ls(params)
+  } catch (e) {
+    if (e.toString().includes('429 Too Many Requests')) {
+      backoffs[service] = {
+        lastTry: new Date().getTime(),
+        tryAfter: tryAfter * 3
```

Comment:
We should re-use the logic in pinning service compliance checker to prevent a 3x increase in retry times.